bin/knc.c

```diff
@@ -47,6 +55,10 @@
 #include <time.h>
 #include <unistd.h>
 
+#ifndef POLLRDHUP
+#define POLLRDHUP 0
+#endif
+
```

Comment:
Maybe:

```
#ifndef EWOULDBLOCK
#define EWOULDBLOCK EAGAIN
#endif
```

then check only for `EWOULDBLOCK` in the code.  That would maximize portability.

Reply:
According to the spec, EWOULDBLOCK can be defined and different from EAGAIN, so I can not take that suggestion. I'll add a check for EWOULDBLOCK for portability.

I pushed another commit adding checks for EWOULDBLOCK in addition to EAGAIN.